### ISSUE TITLE
Error loading Alertmanager config: data key not found

### ISSUE THREAD

**Issue by jvassev** [grafana/grafana#56506]
**What happened**:
I configured an alert (on the built-in grafana alertmanager). Also added a pagerduty contact point.
Once the alerts start firing I can no longer see neither `Contact points` nor `Notification policies` - error is "Error loading Alertmanager config"

**What you expected to happen**:
Receive notification on pagerduty ( I have tested that PD integration works using "Test" button)

**How to reproduce it (as minimally and precisely as possible)**:
1. create a pagerdity contact point,
2. create any rule
3. wait for the rule to fire
4. try listing the alerts in the browser
5. Expect this error message:
```
Error loading Alertmanager config
failed to decrypt stored secure setting: data key not found: failed to decrypt stored secure setting: data key not found
```
![grafana-error](https://user-images.githubusercontent.com/283438/194409274-5262eda4-9482-4aa8-841f-6535f8d30208.png)

**Anything else we need to know?**:
I think I tracked it down to this line https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/notifier/alertmanager_config.go#L56

I had to delete a few rows from the table `alert_configuration` to make the UI usable again. Same problem when I create a new alert.

I've migrated from 8.x recently with no issues (I had not used alerts yet).

I also captured an alert from the database:
```json
{
    "template_files": {},
    "alertmanager_config": {
        "route": {
            "receiver": "grafana-default-email"
        },
        "templates": null,
        "receivers": [
            {
                "name": "grafana-default-email",
                "grafana_managed_receiver_configs": [
                    {
                        "uid": "hZB1Uw4Vz",
                        "name": "email receiver",
                        "type": "email",
                        "disableResolveMessage": false,
                        "settings": {
                            "addresses": "\u003cexample@email.com\u003e"
                        },
                        "secureSettings": null
                    }
                ]
            },
            {
                "name": "test-pd",
                "grafana_managed_receiver_configs": [
                    {
                        "uid": "EZf18Q4Vz",
                        "name": "test-pd",
                        "type": "pagerduty",
                        "disableResolveMessage": false,
                        "settings": {
                            "class": "the-class",
                            "component": "grafana",
                            "group": "grp",
                            "severity": "info",
                            "summary": "summary-template"
                        },
                        "secureSettings": {
                            "integrationKey": "xxxxxxxxxxxx"
                        }
                    }
                ]
            }
        ]
    }
}
```
**Environment**:
- Grafana version: 9.1.5
- Data source type & version: prometheus
- OS Grafana is installed on: containerized
- User OS & Browser: linux/chrome
- Grafana plugins: default ones
- Others:

**Comment by grobinson-grafana** [comment]
Hi! :wave: Do you know if you created the Pagerduty contact point before or after the migration to Grafana 9?

**Comment by grobinson-grafana** [comment]
Hi! 👋 Do you remember if you created the Pagerduty contact point before or after the migration to Grafana 9? Thanks!

**Comment by jvassev** [comment]
Hi, I created all objects after migrating to 9.1.5

**Comment by grobinson-grafana** [comment]
Hi! :wave: I've followed the instructions in **How to reproduce it**  but I'm afraid I didn't get the same error, it all worked as expected. Could you delete and create your Pagerduty contact point a second time and see if the error still persists?

**Comment by jvassev** [comment]
Hi!,
I am getting the same error. I tried upgrading to `v9.1.6 (92461d8d1e)` - issue remains.

Could it have to do with the authentication that I use for grafana?

**Comment by jvassev** [comment]
I've also found this which seems related:
```
logger=ngalert.multiorg.alertmanager t=2022-10-13T14:58:35.158325604Z level=error msg="failed to apply Alertmanager config for org" org=57 id=133 err="unable to apply configuration: failed to build integration map: the receiver is invalid: failed to validate receiver \"pduty\" of type \"pagerduty\": could not find integration key property in settings"
```
The record for id=133 looks like this:
```json
{
  "template_files": {},
  "alertmanager_config": {
    "route": {
      "receiver": "pduty"
    },
    "templates": null,
    "receivers": [
      {
        "name": "grafana-default-email",
        "grafana_managed_receiver_configs": [
          {
            "uid": "CMoZ7lV4z",
            "name": "email receiver",
            "type": "email",
            "disableResolveMessage": false,
            "settings": {
              "addresses": "\u003cexample@email.com\u003e"
            },
            "secureSettings": null
          }
        ]
      },
      {
        "name": "pduty",
        "grafana_managed_receiver_configs": [
          {
            "uid": "Z7TZnlV4k",
            "name": "pduty",
            "type": "pagerduty",
            "disableResolveMessage": false,
            "settings": {},
            "secureSettings": {
              "integrationKey": "xxxxxx"
            }
          }
        ]
      }
    ]
  }
}
```

**Comment by alexmobo** [comment]
Hi @jvassev, we have finally been able to reproduce the error. Could you please check if your `data_keys` table is empty in the database?

**Comment by alexmobo** [comment]
We have found that the error you are experiencing is related to a non-existing row in `data_keys` linked to the existing pagerduty contact point. We have been able to solve the issue by removing the contact point from the json saved in `alert_configuration` table and recreating the contact point after that. Let us know if that works for you.

**Comment by jvassev** [comment]
@alexmobo My data_keys had a few entries but was missing the ones refered to in the alert_configuration json.
Also, i fixed it the same way you described - by deleting rows directly as the UI stopped working.

**Comment by yuwaMSFT2** [comment]
Any idea on what could have caused this? Is this missing reference only happening to a specific type of contact point (pagerduty) and only to Alert configuration? Thanks!

**Comment by joanlopez** [comment]
Hey @grafana/alerting-squad-backend, did you finally find the root cause of these issues? Once having found a possible (manual) workaround, would be good to understand better what happened. I'm still wondering what could have happened and if that's a "past" thing (like a data migration that went wrong, or something like that) or if it's something that's still happening / reproducible on the latest Grafana version and with fresh data.

Thanks!

**Comment by alexmobo** [comment]
Hi @joanlopez, we didn't find the root cause as we couldn't reproduce the cause for the data key's disappearance. Migrations worked fine, tested every possibility I could think of related to alerting migrations.

Nothing threw the slightest clue on what could have happened.

**Comment by mysterious-ben** [comment]
I've experienced the same error "Error loading Alertmanager config: data key not found". I'm using Grafana 9.2.3. I haven't migrated - this is the version I've used from the start. Unfortunately, I'm not sure what caused this problem. It'd been working fine for two months, and then this error appeared (possibly, after a configuration update via Grafana API).

Grafana version: 9.2.3
Data source type & version: PostreSQL, Loki
OS Grafana is installed on: Ubuntu 20.04.5 LTS
User OS & Browser: Pop_OS! & Firefox
Grafana plugins: default ones

**Comment by joanlopez** [comment]
> I've experienced the same error "Error loading Alertmanager config: data key not found". I'm using Grafana 9.2.3. I haven't migrated - this is the version I've used from the start. Unfortunately, I'm not sure what caused this problem. It'd been working fine for two months, and then this error appeared (possibly, after a configuration update via Grafana API).
> 
> Grafana version: 9.2.3 Data source type & version: PostreSQL, Loki OS Grafana is installed on: Ubuntu 20.04.5 LTS User OS & Browser: Pop_OS! & Firefox Grafana plugins: default ones

Hi @mysterious-ben,

I'm sorry to read it, but thanks for sharing the details.
We identified a potential source of such issues (which fits what you mentioned -after configuration update-) and made a [temporary fix](https://github.com/grafana/grafana/pull/60664) that'll be shipped in v9.4.
More improvements will come (maybe by v9.5) but at least this should reduce a lot the chances from that happening again.

Thanks in advance for your patience!
